### DISCUSSION TITLE
Do we still need mod_search?

### DISCUSSION THREAD

**Discussion by Worti2** [joomla/joomla-cms#38303] [General] [answered]
Hello everybody,

very many Joomla! users use the mod_search module, because in Joomla! 2.5 and Joomla! 3 you could not really use mod_finder. When they upgrade their site to Joomla! 4, they still use it. But the code is the old one, e.g. `class="element-invisible"` or `<input type="image" alt="' . $button_text . '" class="button" src="' . $img . '" onclick="this.form.searchword.focus();"/>`.

And now the discussion:

Do we still need or want mod_search?

Case "Yes": Please rework the code: e.g. swap image for icon and change the class to visually-hidden.

Case "No": Please decouple mod_search from the Joomla! 4 core and leave it out completely in Joomla! 5. Then I help myself with an override until then.

Greetings
Markus

**Comment by brianteeman** [comment, marked as answer]
Update search to the latest version (its now an uncoupled extension) and the coponent and module will be updated. If there are still issues with the module then they should be addressed in its own repo which is https://github.com/joomla-extensions/search

**Comment by Worti2** [comment]
Hi Brian,

I will make some suggestions there.

Thank you and

greetings
Markus

**Comment by Hackwar** [comment]
Closing, since this has been solved.
